File: app/src/main/java/com/github/kirillf/oauth/github/Followers.java
package com.github.kirillf.oauth.github;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.util.Arrays;
import java.util.List;

/**
 * Copyrigh Mail.ru Games (c) 2015
 * Created by y.bereza.
 */

public class Followers {
    private List<User> users;

    public static Followers createFromJSON(String json) {
        Gson gson = new GsonBuilder().create();
        Followers followers = new Followers();
        followers.users = Arrays.asList(gson.fromJson(json, User[].class));

        return followers;
    }

    public String asJSON() {
        Gson gson = new GsonBuilder()
                .setPrettyPrinting()
                .create();
        return gson.toJson(users);
    }

    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (User user : users) {
            sb.append("name: ");
            sb.append(user.getLogin());
            sb.append("\n");
            sb.append("type: ");
            sb.append(user.getType());
            sb.append("\n");
            sb.append("admin: ");
            sb.append(user.getSiteAdmin());
            sb.append("\n");
        }
        return sb.toString();
    }
}

File: app/src/main/java/com/github/kirillf/oauth/github/Followers.kt
package com.github.kirillf.oauth.github

import com.google.gson.GsonBuilder

class Followers {
    private var users: Array<User>? = null

    fun asJSON(): String {
        val gson = GsonBuilder()
                .setPrettyPrinting()
                .create()
        return gson.toJson(users)
    }

    override fun toString(): String {
        val sb = StringBuilder()
        for (user in users!!) {
            sb.append("name: ")
            sb.append(user.login)
            sb.append("\n")
            sb.append("type: ")
            sb.append(user.type)
            sb.append("\n")
            sb.append("admin: ")
            sb.append(user.siteAdmin)
            sb.append("\n")
        }
        return sb.toString()
    }

    companion object {

        fun createFromJSON(json: String): Followers {
            val gson = GsonBuilder().create()
            val followers = Followers()
            followers.users = gson.fromJson(json, Array<User>::class.java) as Array<User>

            return followers
        }
    }
}
